### PR TITLE
feat: add Stepfun Model Provider to Casibase

### PR DESCRIPTION
**Pull request by stardustlove**
Part of: https://github.com/casibase/casibase/issues/819

### CODE REVIEW

**Comment by stardustlove** [comment]
feat: add Stepfun Model Provider to Casibase #819

File: model/stepfun.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+			Messages: []ChatCompletionMessage{user_prompt},
+		}
+	}
+
+	requestBodyJSON, err := json.Marshal(messages)
+	if err != nil {
+		log.Fatalf("JSON Marshal Error: %v", err)
+	}
+
+	url := "https://api.stepfun.com/v1/token/count"
```
Comment (hsluoyz) [inline model/stepfun.go]: Why call another API to calculate tokens? Their docs (https://platform.stepfun.com/docs/pricing/intro) said, they already returned the token usage in the response?

image

File: web/src/Setting.js [hunk 2]
```diff
@@ -1006,6 +1007,15 @@ export function getProviderSubTypeOptions(category, type) {
     return ([
       {id: "Dummy", name: "Dummy"},
     ]);
+  } else if (type === "StepFun") {
```
Comment (hsluoyz) [inline web/src/Setting.js:1011]: Put before Dummy

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@stardustlove resolve the conflicts

image

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**